src/local_storage.rs

```diff
@@ -69,8 +71,10 @@ impl LocalStorage {
         self.storage
             .get(&(participant_id, TypeId::of::<T>()))
             .map(|any| {
-                any.downcast_ref::<T::Value>()
-                    .ok_or(InternalError::InternalInvariantFailed)
+                any.downcast_ref::<T::Value>().ok_or_else(|| {
+                    error!("Could not downcast storage entry to <T as TypeTag>::Value");
```

Comment:
suggestion: We should log the `participant_id` and [`T`](https://stackoverflow.com/a/58119924) here. We might also log `any`, but I'm not sure if it'll be atrociously long and unhelpful.